int32 user_id
int32 location
---
bool direction
